### ISSUE TITLE
[backend] Failed to retry run: Workflow cannot be retried with node in Omitted phase

### ISSUE THREAD

**Issue by caleb-artifact** [kubeflow/pipelines#5962]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
Kubeflow Pipeline Standalone via kustomize manifests on GKE

*  KFP version: 
1.6.0

*  KFP SDK version: 
```
kfp                           1.6.3
kfp-pipeline-spec             0.1.8
kfp-server-api                1.6.0
```

### Steps to reproduce

1. Run a pipeline with a component depending upon another failing component
2. Retry the run

```jsonc
//  Failed to retry run: 5b79857a-be4f-4876-98ae-105845e892ee with error:
{
   "error": "Retry run failed.: InternalServerError: Workflow cannot be retried with node nlp-pipeline-mcclt-4094140295 in Omitted phase: workflow cannot be retried",
   "code": 13,
   "message": "Retry run failed.: InternalServerError: Workflow cannot be retried with node nlp-pipeline-mcclt-4094140295 in Omitted phase: workflow cannot be retried",
   "details": [{
      "@type": "type.googleapis.com/api.Error",
      "error_message": "Internal Server Error",
      "error_details": "Retry run failed.: InternalServerError: Workflow cannot be retried with node nlp-pipeline-mcclt-4094140295 in Omitted phase: workflow cannot be retried"
   }]
}
```

### Expected result

The pipeline is retreid.

### Materials and Reference

I noticed this bug was fixed in 1.5.1 already but hasn't been fixed in 1.6.x.

I tired to get some info on when a fix would land in a  1.6 release but have not seen any activity there since the issue was closed.

https://github.com/kubeflow/pipelines/issues/5878

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by capri-xiyue** [comment]
As https://github.com/kubeflow/pipelines/issues/5878#issuecomment-863651456 mentioned, you need to wait for next KFP release 1.7.0.

**Comment by Bobgy** [comment]
I think we can close now

**Comment by caleb-artifact** [comment]
> As [#5878 (comment)](https://github.com/kubeflow/pipelines/issues/5878#issuecomment-863651456) mentioned, you need to wait for next KFP release 1.7.0.

Ah got ya.  I think what confused me is that v1.5.1 included the fix and I had linked that comment to the 1.5.1 release because the author was using 1.5.0 and thus 1.5.1 would be the "next release" for their KFP version.
